### PR TITLE
[rl][bug] Fix gradient accumulation zero grad

### PR DESCRIPTION
**Pull request by felipemello1**
We call it at the wrong place (fwd/bwd instead of optimizer step), zeroing grad from all micro-batches except the last one, i.e. no accumulation

### CODE REVIEW

**Review by wwwjn — APPROVED** [review]

**Review by tianyu-l — CHANGES_REQUESTED** [review]
it's subtle but core trainer calls it at beginning
https://github.com/pytorch/torchtitan/blob/main/torchtitan/trainer.py#L707

**Comment by felipemello1** [comment]
> it's subtle but core trainer calls it at beginning [`main`/torchtitan/trainer.py#L707](https://github.com/pytorch/torchtitan/blob/main/torchtitan/trainer.py#L707)

the APIs are different though. We call fwd_bwd directly, and then .optimize. The only way to call at the beginning is if the controller call it **before** the micro batches start, e.g.

```python
# Controller 

trainer.zero_grads.call()
for microbatch in microbatches:
	trainer.fwd_bwd.call(microbatch)
trainer.optimizer_step.call()
```

I dont think that we want to add the API zero_grads, since we are following the tinker API, which is standard in most RL libraries now
```
forward_backward()
optim_step()
```

@tianyu-l 

let me know if you disagree!

**Review by tianyu-l — APPROVED** [review]
sg
